### ISSUE TITLE
Toolstripmenuitem under menubar cannot clearly distinguish checkstate is checked or unchecked

### ISSUE THREAD

**Issue by MelonWang1** [dotnet/winforms#9173]
### .NET version

.NET 8.0.100-preview.5.23274.2

### Did it work in .NET Framework?

No

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

No, this issue also repro in .NET7.0 & .NET 6.0 & .NET 5.0 & .NET 3.1.

### Issue description

Toolstripmenuitem under menubar cannot clearly distinguish checkstate is checked or unchecked

Toolstripbutton under menubar can learly distinguish checkstate is checked or unchecked, checked state has rectangle box.
![toolstripmenuitem](https://github.com/dotnet/winforms/assets/94418985/4d59b139-7c7d-406f-9356-3e350167ebc5)

Toolstripmenuitem under menu can clearly distinguish checkstate is checked or unchecked, checked state has check symbol.
![Screenshot 2023-05-25 081546](https://github.com/dotnet/winforms/assets/94418985/ed0dc811-4660-4297-9a33-4ba381df9cb4)

### Steps to reproduce

1. Create a Winforms .net core application with MenuStrip control
2. Add 2 menuitem under menubar
3. Change toolstripmenuitem2 checkstate property is checked
4. Build and run application
5. Observe toolstripmenuitem2

or use this sample application:
[test.zip](https://github.com/dotnet/winforms/files/11562500/test.zip)

**Comment by elachlan** [comment]
@MelonWang1 is this fixed?

**Comment by MelonWang1** [comment]
No, still repro on latest .NET 8.0 SDK: 8.0.100-preview.7.23364.32.

**Comment by elachlan** [comment]
Are we expecting that the `Toolstripmenuitem`shows a similar state as the `Toolstripbutton`? or should it have a check mark against it?
